tok/hnsw/helper.go

```diff
@@ -518,6 +519,56 @@ func (ph *persistentHNSW[T]) newPersistentEdgeKeyValueEntry(ctx context.Context,
 	return edge, nil
 }
 
+func (ph *persistentHNSW[T]) distance_betw(ctx context.Context, tc *TxnCache, inUuid, outUuid uint64, inVec,
+	outVec *[]T) T {
+	err := ph.getVecFromUid(outUuid, tc, outVec)
+	if err != nil {
+		log.Printf("[ERROR] While getting vector %s", err)
+		return -1
+	}
+
+	d, err := ph.simType.distanceScore(*inVec, *outVec, ph.floatBits)
+	if err != nil {
+		log.Printf("[ERROR] While getting vector %s", err)
+		return -1
+	}
+	return d
+}
+
+type CustomHeap struct {
```

Comment:
can you rename this so its a little more clear what its doing. is this a custom heap of all neighbors of a given node at a given layer? or is it essentially just holding a tuple and a compare function?